### PR TITLE
LOK-2478 :Syslog message from Camel route to Syslog SinkModule

### PR DESCRIPTION
**Pull request by MusaidAli**
## Description
Syslog message from camel route to Syslog SinkModule

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2478

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog.proto [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import "snmp-api.proto";
+import "opennms_minion_ipc.proto";
+
+
+message SyslogIdentity {
+  int32 generic = 1;
+  int32 specific = 2;
+  string enterpriseId = 3;
+  string syslogOID = 4;
```
Comment (fooker) [inline shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog.proto:16]: What do these fields mean?

File: shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog.proto [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+message SyslogMessageDTO {
+  string syslogAddress = 1;
+  string port = 2 ;
+  string systemId = 3;
+  string location = 4 ;
+  bytes rawMessage = 5;
+  string community = 6;
```
Comment (fooker) [inline shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog.proto:25]: What is the `community`?

File: shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog.proto [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  string version = 7;
```
Comment (fooker) [inline shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog.proto:26]: What is the `version`?

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java [hunk 4]
```diff
@@ -77,6 +108,46 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
...
+                            LOG.info("Host:" + m_message.getHostname());
+                            LOG.info("Date:" + m_message.getDate().toString());
+
+                            SyslogIdentity syslogIdentity = SyslogIdentity.newBuilder()
+                                    .setEnterpriseId("EntID")
+                                    .setGeneric(1)
+                                    .build();
+                            Identity identity1 = Identity.newBuilder()
+                                    .setSystemId("SyslogID")
+                                    .build();
```
Comment (fooker) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java:129]: I'm pretty sure this is non-sense.

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java [hunk 5]
```diff
@@ -77,6 +108,46 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
...
+                        try {
+                            String content = buffer.toString(StandardCharsets.UTF_8);
+                            LOG.info("content" + content);
+                            SyslogMessage m_message = syslogParser.parse(content);
+                            LOG.info("Severity:" + m_message.getSeverity());
+
+                            LOG.info("Message:" + m_message.getMessage());
+                            LOG.info("Facility:" + m_message.getFacility());
+                            LOG.info("Host:" + m_message.getHostname());
+                            LOG.info("Date:" + m_message.getDate().toString());
```
Comment (fooker) [inline minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java:121]: Do not info-log every message

File: events/syslog/src/main/java/org/opennms/horizon/events/syslog/impl/SysLogProtoToEventLogXmlMapperImpl.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    @Override
+    public Log convert(TenantLocationSpecificSyslogLogDTO messageLog) {
+        Log log = new Log();
+        Events events = new Events();
+        log.setEvents(events);
+
+        String tenantId = messageLog.getTenantId();
+        String locationId = messageLog.getLocationId();
+
+        // TODO: Add metrics for Traps received/error/dropped.
```
Comment (fooker) [inline events/syslog/src/main/java/org/opennms/horizon/events/syslog/impl/SysLogProtoToEventLogXmlMapperImpl.java:60]: ??

**Review by fooker — CHANGES_REQUESTED** [review]

File: shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog.proto (hunk 1 above)
Comment (MusaidAli) [reply shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog.proto]: removed ,

File: shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog.proto (hunk 2 above)
Comment (MusaidAli) [reply shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog.proto]: removed community ,community public belongs to snmp

File: shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog.proto (hunk 3 above)
Comment (MusaidAli) [reply shared-lib/lokahi-ipc/ipc-grpc/ipc-grpc-contract/src/main/proto/opennms/syslog/syslog.proto]: removed ,version

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java (hunk 5 above)
Comment (MusaidAli) [reply minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java]: refactored.

File: events/syslog/src/main/java/org/opennms/horizon/events/syslog/impl/SysLogProtoToEventLogXmlMapperImpl.java (hunk 6 above)
Comment (MusaidAli) [reply events/syslog/src/main/java/org/opennms/horizon/events/syslog/impl/SysLogProtoToEventLogXmlMapperImpl.java]: removed //TODO

File: minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java (hunk 4 above)
Comment (MusaidAli) [reply minion/minion-syslog/listener/src/main/java/org/opennms/horizon/minion/syslog/listener/SyslogRoute.java]: yeah,it is removed.

**Pull request closed without merging**